simul/platform/aws.go

```diff
@@ -0,0 +1,278 @@
+	//GOOS=linux GOARCH=amd64 go build
+	os.Setenv("GOOS", a.targetSystem)
+	os.Setenv("GOARCH", a.targetArch)
+	cmd := NewCommand("go", "build", "-o", binPath, path)
+
+	if err := cmd.Run(); err != nil {
+		fmt.Println("stdout -> " + cmd.ReadAll())
+		return err
+	}
+	return nil
+}
+
+func (a *awsPlatform) Configure(c *lib.Config) error {
+
+	CMDS := aws.NewCommands("/tmp/masterAWS", "/tmp/nodeAWS", "/tmp/aws.conf", "/tmp/aws.csv")
+	a.masterCMDS = aws.MasterCommands{CMDS}
+	a.slaveCMDS = aws.SlaveCommands{CMDS}
+
+	// Compile binaries
+	a.pack("github.com/ConsenSys/handel/simul/node", c, CMDS.SlaveBinPath)
+	a.pack("github.com/ConsenSys/handel/simul/master", c, CMDS.MasterBinPath)
+
+	// write config
+	if err := c.WriteTo(CMDS.ConfPath); err != nil {
+		return err
+	}
+
+	//Start EC2 instances
+	if err := a.aws.StartInstances(); err != nil {
+		return err
+	}
+
+	// Create master and slave instances
+	masterInstance, slaveInstances, err := makeMasterAndSlaves(a.aws.Instances())
+	if err != nil {
+		fmt.Println(err)
+		return err
+	}
+
+	cons := c.NewConstructor()
+	masterAddr := aws.GenRemoteAddress(*masterInstance.PublicIP, 5000)
+	masterNode := lib.GenerateNode(cons, -1, masterAddr)
+	nodeAndSync := aws.NodeAndSync{masterNode, ""}
+	//Create master controller
+	master, err := aws.NewSSHNodeContlorrer(nodeAndSync, a.pemBytes, a.user)
+	if err != nil {
+		return err
+	}
+	a.master = master
+
+	for {
+		err := master.Init()
+		if err != nil {
+			fmt.Println("Master Init failed, trying one more time", err, *masterInstance.ID, *masterInstance.PublicIP, *masterInstance.State)
+			time.Sleep(3 * time.Second)
+			continue
+		}
+		break
+	}
+
+	fmt.Println("[+] Master Instances")
+	fmt.Println("	 [-] Instance ", *masterInstance.ID, *masterInstance.State, masterAddr)
+	fmt.Println()
+	fmt.Println("[+] Avaliable Slave Instances:")
+
+	for i, inst := range slaveInstances {
+		fmt.Println("	 [-] Instance ", i, *inst.ID, *inst.State, *inst.PublicIP)
+	}
+
+	fmt.Println("[+] Transfering files to Master:", CMDS.MasterBinPath, CMDS.SlaveBinPath, CMDS.ConfPath)
+	master.CopyFiles(CMDS.MasterBinPath, CMDS.SlaveBinPath, CMDS.ConfPath)
+	configure := a.masterCMDS.Configure()
+
+	//*** Configure Master
+	fmt.Println("[+] Configuring Master")
+	for idx := 0; idx < len(configure); idx++ {
+		fmt.Println("       Exec:", idx, configure[idx])
+		_, err := master.Run(configure[idx])
+		if err != nil {
+			return err
+		}
+	}
+
+	//*** Configure Slaves
+	slaveCmds := a.slaveCMDS.Configure(*masterInstance.PublicIP)
+	fmt.Println("")
+	fmt.Println("")
+	fmt.Println("[+] Configuring Slaves:")
+
+	addresses, syncs := aws.GenRemoteAddresses(slaveInstances)
+	var wg sync.WaitGroup
+
+	for i, addr := range addresses {
+		node := lib.GenerateNode(cons, i, addr)
+		nodeAndSync := aws.NodeAndSync{node, syncs[i]}
+		wg.Add(1)
+		// TODO This might become a problem for large number of slaves,
+		// limit numebr of go-routines running concurrently if this is the case
```

Comment:
even 5000 go routine on our laptops should be fine - these are not CPU intensives so I think it's ok but let's see

Reply:
I agree